asserts/device_asserts.go

```diff
@@ -156,6 +156,10 @@ func checkSnapWithTrackHeader(header string, headers map[string]interface{}) err
 		return fmt.Errorf(`%q header must be a string`, header)
 	}
 	l := strings.SplitN(value, "=", 2)
+
+	if err := validateSnapName(l[0], header); err != nil {
```

Comment:
Would it make sense to move this below `if len(l) == 1 {` check?

Reply:
No, we always want to validate the name, even if there's no track.

Reply:
Ah, got you, thanks.